File: solutions/ruby/sum-of-multiples/1/sum_of_multiples.rb
=begin
Write your code for the 'Sum Of Multiples' exercise in this file. Make the tests in
`sum_of_multiples_test.rb` pass.

To get started with TDD, see the `README.md` file in your
`ruby/sum-of-multiples` directory.
=end

class SumOfMultiples
  def initialize(*numbers)
    @numbers = numbers
  end

  def to(limit)
    mult = Array.new
    if @numbers[0] == 0 or @numbers == []
      return 0
    end
    for x in 1..(limit - 1)
      for f in @numbers
        if x % f == 0
          if !mult.include?(x)
            mult.push(x)
          end
        end
      end
    end
    if mult.length == 0
      0
    else
      mult.reduce(0, :+)
    end
  end
end

File: python/sum-of-multiples/sum_of_multiples.py
def sum_of_multiples(limit: int, multiples: list[int]) -> int:
    set_of_multiples = set([x for x in range(1, limit) if any(x % m == 0 for m in multiples if m != 0)])
    print(set_of_multiples)
    return sum(set_of_multiples)
